### PR TITLE
RavenDB-5773 Implement LazySessionOperations for the new client - WIP

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs [hunk 1]
```diff
@@ -96,39 +96,48 @@ public void HandleResponses(GetResponse[] responses, ShardStrategy shardStrategy
 */
         }
 
-        public void HandleResponse(GetResponse response)
+        public void HandleResponse(BlittableJsonReaderObject response)
         {
-            throw new NotImplementedException();
 
-            /*if (response.ForceRetry)
+            object forceRetry;
```
Comment (ayende) [inline src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs:102]: Why isn't this a boolean?

File: src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs [hunk 2]
```diff
@@ -96,39 +96,48 @@ public void HandleResponses(GetResponse[] responses, ShardStrategy shardStrategy
...
+            if (( forceRetry!= null) && ((bool)forceRetry))
             {
                 Result = null;
                 RequiresRetry = true;
                 return;
             }
 
-            var result = response.Result;
+            object result;
+            response.TryGetMember("Result", out result);
```
Comment (ayende) [inline src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs:113]: Why isn't this a `BlittableJsonReaderObject`?

File: src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs [hunk 3]
```diff
@@ -96,39 +96,48 @@ public void HandleResponses(GetResponse[] responses, ShardStrategy shardStrategy
...
-        }*/
+                  Result = loadOperation.GetDocuments<T>();
+        }
 
         public IDisposable EnterContext()
         {
-            throw new NotImplementedException();
-            /* return loadOperation.EnterLoadContext();*/
+            return null;
+            //throw new NotImplementedException();
```
Comment (ayende) [inline src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs:139]: why is this commented?

File: src/Raven.NewClient/Commands/MultiGetCommand.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            foreach (var req in GetCommands)
+            {
+                var headers = new DynamicJsonValue();
+                foreach (var header in req.Headers)
+                {
+                    headers[header.Key] = header.Value;
+                }
+                commands.Add(new DynamicJsonValue()
+                {
+                    ["Url"] = $"/databases/{node.Database}{req.Url}",
```
Comment (ayende) [inline src/Raven.NewClient/Commands/MultiGetCommand.cs:37]: What about includes?

File: src/Raven.NewClient/Commands/MultiGetCommand.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                        if (!(first))
+                            writer.WriteComma();
+                        first = false;
+                        Context.Write(writer, command);
+                    }
+                    writer.WriteEndArray();
+                }
+            });
+
+            var sb = new StringBuilder($"{node.Url}/databases/{node.Database}/multi_get?parallel=yes");
```
Comment (ayende) [inline src/Raven.NewClient/Commands/MultiGetCommand.cs:62]: We don't need the parallel any longer, unused.

File: src/Raven.NewClient/Document/Batches/LazyMultiLoaderWithInclude.cs [hunk 6]
```diff
@@ -10,7 +10,7 @@ namespace Raven.NewClient.Client.Document.Batches
     public class LazyMultiLoaderWithInclude<T> : ILazyLoaderWithInclude<T>
     {
         private readonly IDocumentSessionImpl session;
-        private readonly List<KeyValuePair<string, Type>> includes = new List<KeyValuePair<string, Type>>();
+        private readonly List<string> includes = new List<string>();
```
Comment (ayende) [inline src/Raven.NewClient/Document/Batches/LazyMultiLoaderWithInclude.cs:13]: are you sure we won't need the type info here?

File: src/Raven.NewClient/Document/DocumentSession.cs [hunk 7]
```diff
@@ -159,7 +163,100 @@ protected override Task<string> GenerateKeyAsync(object entity)
...
+
+            try
+            {
+                var sw = Stopwatch.StartNew();
+
+                IncrementRequestCount();
+
+                var responseTimeDuration = new ResponseTimeInformation();
+
+                while (ExecuteLazyOperationsSingleStep(responseTimeDuration))
```
Comment (ayende) [inline src/Raven.NewClient/Document/DocumentSession.cs:177]: why do we need to do this?

File: src/Raven.NewClient/Document/DocumentSession.cs [hunk 8]
```diff
@@ -159,7 +163,100 @@ protected override Task<string> GenerateKeyAsync(object entity)
...
+                        case 304:
+                        case 404:
+                            break;
+                        default:
+                            throw new InvalidOperationException("Got an error from server, status code: " + (int)status +
+                                                            Environment.NewLine + response);
+                    }
+
+                    pendingLazyOperations[i].HandleResponse(response);
+                    if (pendingLazyOperations[i].RequiresRetry)
```
Comment (ayende) [inline src/Raven.NewClient/Document/DocumentSession.cs:246]: what scenario will generate this?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs (hunk 1 above)
Comment (efratshenhar) [reply src/Raven.NewClient/Commands/Lazy/LazyLoadOperation.cs:102]: TryGetMember can return only object , I will change all to TryGet

**Pull request merged**